packages/polaris-viz/src/components/shared/FunnelChartSegment/components/ScaledSegment.tsx

```diff
@@ -0,0 +1,163 @@
+  x,
+  onMouseEnter,
+  onMouseLeave,
+  children,
+}: Props) {
+  const {containerBounds} = useChartContext();
+  const {width: drawableWidth, height: drawableHeight} = containerBounds ?? {
+    height: 0,
+    width: 0,
+  };
+  const topSegmentHeight = calculateTopSegmentHeight(barHeight);
+  const isFirst = index === 0;
+  const dimensions: Dimensions = {
+    width: barWidth,
+    height: barHeight,
+    x,
+    y: drawableHeight - barHeight,
+  };
+
+  const topSegmentMarkup = (
+    <path
+      d={getRoundedRectPath({
+        width: barWidth,
+        height: topSegmentHeight,
+        borderRadius: `${isFirst ? FUNNEL_SEGMENT.borderRadius : 0} ${
+          isLast ? FUNNEL_SEGMENT.borderRadius : 0
+        } 0 0`,
+      })}
+      fill={FUNNEL_SEGMENT.colors.primary}
+    />
+  );
+
+  const scaleSize = calculateResponsiveScale(drawableHeight, drawableWidth);
+  const calculateScaleHeight = () =>
+    scaleSize * FUNNEL_SEGMENT.scaleHeightMultiplier;
+
+  const scaleEffectMarkup = (
+    <g transform={`translate(0, ${topSegmentHeight})`}>
+      {[
+        FUNNEL_SEGMENT.colors.scaleLight,
+        FUNNEL_SEGMENT.colors.scaleShadow,
+        FUNNEL_SEGMENT.colors.scaleLight,
+        FUNNEL_SEGMENT.colors.scaleShadow,
+      ].map((fill, index) => (
+        <rect
+          key={`scale-effect-${index}`}
+          x={0}
+          y={scaleSize * index}
+          width={barWidth}
+          height={scaleSize}
+          fill={fill}
+        />
+      ))}
+    </g>
+  );
+
+  const scaleRippleMarkup = (
+    <g transform={`translate(0, ${topSegmentHeight})`}>
+      <path
+        d={`M ${scaleSize * 1.5},${scaleSize} L 0,0 L 0,${scaleSize * 2} Z`}
+        fill={FUNNEL_SEGMENT.colors.ripple}
+      />
+      <path
+        d={`M ${scaleSize * 1.5},${scaleSize * 3} L 0,${scaleSize * 2} L 0,${
+          scaleSize * 4
+        } Z`}
+        fill={FUNNEL_SEGMENT.colors.ripple}
+      />
+    </g>
+  );
+
+  const bottomSegmentMarkup = (
+    <rect
+      x={0}
+      y={topSegmentHeight + calculateScaleHeight()}
+      width={barWidth}
+      height={barHeight - topSegmentHeight - calculateScaleHeight()}
+      fill={FUNNEL_SEGMENT.colors.primary}
+    />
+  );
+
+  return (
+    <Fragment>
+      <g transform={`translate(${dimensions.x}, ${dimensions.y})`}>
+        {topSegmentMarkup}
+        {scaleEffectMarkup}
+        {scaleRippleMarkup}
+        {bottomSegmentMarkup}
+        <InteractiveOverlay
+          width={barWidth}
+          height={barHeight}
+          index={index}
+          onMouseEnter={onMouseEnter}
+          onMouseLeave={onMouseLeave}
+        />
+      </g>
+      {children}
+    </Fragment>
+  );
+}
+
+const calculateTopSegmentHeight = (height: number) =>
+  Math.floor(height * FUNNEL_SEGMENT.topSegmentHeightRatio);
+
+const calculateResponsiveScale = (
+  drawableHeight: number,
+  drawableWidth: number,
+) => {
+  const heightScale = drawableHeight * 0.015;
+  const widthScale = drawableWidth * 0.005;
```

Comment:
Can we use variables for these numbers? I don't know what they represent and would be lost if I had to alter this